src/Hazelcast.Net/Core/ClientVersion.cs

```diff
@@ -59,18 +59,38 @@ internal static string GetVersion(AssemblyInformationalVersionAttribute informat
         /// <summary>
         /// (for tests only)
         /// Gets the major.minor version of a SemVer-compliant version.
-        /// Outside of tests, prefer the <see cref="MajorMinorVersion"/> property.
+        /// Outside of tests, prefer the <see cref="MajorMinorPatchMajorMinorPatchVersion"/> property.
         /// </summary>
         /// <param name="version">The SemVer-compliant version.</param>
         /// <returns>The "pure" version corresponding to the specified <paramref name="version"/>.</returns>
-        internal static string GetMajorMinorVersion(string version)
+        internal static string GetMajorMinorPatchVersion(string version)
         {
             var pos0 = version.IndexOf('.', StringComparison.OrdinalIgnoreCase);
             var pos1 = version.IndexOf('.', pos0 + 1);
-            if (pos1 >= 0)
+            var pos2 = -1;
+            
+            // Check the patch version segment, expecting max 2 digits
+            if (pos1 + 1 < version.Length && int.TryParse(version[pos1 + 1] + "", out _))
             {
-                // two dots = major.minor.whatever
-                // take everything until the second '.' = major.minor
+                pos2 = pos1 + 1; 
+
+                if (pos2 + 1 < version.Length && int.TryParse(version[pos2 + 1] + "", out _))
```

Comment:
do you have unit tests to validate the changes?

Reply:
Yes, please see the `ClientVersionReturnsMajorMinorVersion` in the PR.